### ISSUE TITLE
[ingress-nginx] Controller does not do rolling update correctly 

### ISSUE THREAD

**Issue by name212** [deckhouse/deckhouse#3713]
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Version

1.42

### Expected Behavior

Only one controller pod should unavailable during update.

### Actual Behavior

Multiple pods unavailable during update.
![image](https://user-images.githubusercontent.com/30695496/216001940-b6a3dc61-cc99-4e26-851b-d22f714d4c9e.png)

### Additional Information

Probably need to update to 1.23 kubernetes, on affected clusters we have 1.21

### Logs
ingressnginxcontroller resource
```
apiVersion: deckhouse.io/v1
kind: IngressNginxController
metadata:
  labels:
    app.kubernetes.io/managed-by: Helm
  name: main
spec:
  chaosMonkey: false
  config:
    http-snippet: |
      map $host $application_ingress_access_permitted {
        "example.com" "yes";
        default no;
      }
  controllerVersion: "1.1"
  disableHTTP2: false
  hostPortWithProxyProtocol:
    httpPort: 10080
    httpsPort: 10443
  hsts: false
  ingressClass: nginx-ingress-class
  inlet: HostPortWithProxyProtocol
  maxReplicas: 1
  minReplicas: 1
  nodeSelector:
    dedicated: frontend
  tolerations:
  - effect: NoSchedule
    key: dedicated
    operator: Equal
    value: frontend
  underscoresInHeaders: false
  validationEnabled: false
  waitLoadBalancerOnTerminating: 60
```
Audit logs
```json
{
  "kind": "Event",
  "apiVersion": "audit.k8s.io/v1",
  "level": "Metadata",
  "auditID": "39838b74-49dc-42eb-9ee9-d7ffde49b154",
  "stage": "ResponseComplete",
  "requestURI": "/api/v1/namespaces/d8-ingress-nginx/pods/controller-mainh9jqt",
  "verb": "delete",
  "user": {
    "username": "system:serviceaccount:kube-system:daemon-set-controller",
    "uid": "bbf52e1d-a74d-4689-be5b-35e46e8b0e4c",
    "groups": [
      "system:serviceaccounts",
      "system:serviceaccounts:kube-system",
      "system:authenticated"
    ]
  },
  "sourceIPs": [
    "192.168.101.101"
  ],
  "userAgent": "kube-controller-manager/v1.21.14 (linux/amd64) kubernetes/0000000/system:serviceaccount:kube-system:daemon-set-controller",
  "objectRef": {
    "resource": "pods",
    "namespace": "d8-ingress-nginx",
    "name": "controller-main-h9jqt",
    "apiVersion": "v1"
  },
  "responseStatus": {
    "metadata": {},
    "code": 200
  },
  "requestReceivedTimestamp": "2023-01-31T06:04:26.563032Z",
  "stageTimestamp": "2023-01-31T06:04:26.584110Z",
  "annotations": {
    "authorization.k8s.io/decision": "allow",
    "authorization.k8s.io/reason": "RBAC: allowed by ClusterRoleBinding \"system:controller:daemon-set-controller\" of ClusterRole \"system:controller:daemon-set-controller\" to ServiceAccount \"daemon-set-controller/kube-system\""
  }
}
{
  "kind": "Event",
  "apiVersion": "audit.k8s.io/v1",
  "level": "Metadata",
  "auditID": "bbf3802a-0fcc-44aa-945e-a6ee43de4db0",
  "stage": "ResponseComplete",
  "requestURI": "/api/v1/namespaces/d8-ingress-nginx/pods/controller-main-8d7xq",
  "verb": "delete",
  "user": {
    "username": "system:serviceaccount:kube-system:daemon-set-controller",
    "uid": "bbf52e1d-a74d-4689-be5b-35e46e8b0e4c",
    "groups": [
      "system:serviceaccounts",
      "system:serviceaccounts:kube-system",
      "system:authenticated"
    ]
  },
  "sourceIPs": [
    "192.168.101.101"
  ],
  "userAgent": "kube-controller-manager/v1.21.14 (linux/amd64) kubernetes/0000000/system:serviceaccount:kube-system:daemon-set-controller",
  "objectRef": {
    "resource": "pods",
    "namespace": "d8-ingress-nginx",
    "name": "controller-main-8d7xq",
    "apiVersion": "v1"
  },
  "responseStatus": {
    "metadata": {},
    "code": 200
  },
  "requestReceivedTimestamp": "2023-01-31T06:04:53.430017Z",
  "stageTimestamp": "2023-01-31T06:04:53.494968Z",
  "annotations": {
    "authorization.k8s.io/decision": "allow",
    "authorization.k8s.io/reason": "RBAC: allowed by ClusterRoleBinding \"system:controller:daemon-set-controller\" of ClusterRole \"system:controller:daemon-set-controller\" to ServiceAccount \"daemon-set-controller/kube-system\""
  }
}
{
  "kind": "Event",
  "apiVersion": "audit.k8s.io/v1",
  "level": "Metadata",
  "auditID": "5823c777-dd96-438b-9af4-d3ed1c6b1d49",
  "stage": "ResponseComplete",
  "requestURI": "/api/v1/namespaces/d8-ingress-nginx/pods/controller-main-8d7xq",
  "verb": "delete",
  "user": {
    "username": "system:node:k8s-front-02",
    "groups": [
      "system:nodes",
      "system:authenticated"
    ]
  },
  "sourceIPs": [
    "192.168.101.120"
  ],
  "userAgent": "kubelet/v1.21.14 (linux/amd64) kubernetes/0f77da5",
  "objectRef": {
    "resource": "pods",
    "namespace": "d8-ingress-nginx",
    "name": "controller-main-8d7xq",
    "apiVersion": "v1"
  },
  "responseStatus": {
    "metadata": {},
    "code": 200
  },
  "requestReceivedTimestamp": "2023-01-31T06:11:12.978953Z",
  "stageTimestamp": "2023-01-31T06:11:12.999716Z",
  "annotations": {
    "authorization.k8s.io/decision": "allow",
    "authorization.k8s.io/reason": ""
  }
}

```
```

**Comment by nabokihms** [comment]
@name212  can you also add IngressNginxController resources?

**Comment by name212** [comment]
https://github.com/deckhouse/deckhouse/issues/3713#issuecomment-1411733624

added

**Comment by EvgenySamoylov** [comment]
The root cause is the incorrect numbers in the `status` of the `daemonset`. We have to wait Kubernetes 1.23 for that to be fixed. If that won't help - we will return to that case and either try to play around hook logic for smoother rolling update or try to use another controller, for example https://github.com/DataDog/extendeddaemonset

**Comment by EvgenySamoylov** [comment]
Some part of the issue is addressed here: #3915

**Comment by name212** [comment]
We [tried](https://github.com/DataDog/extendeddaemonset), but controller can not create pod for our ingress-controller.
